cspyce/cspyce1.py

```diff
@@ -810,6 +810,27 @@ def dasec(handle):
 del CSPYCE_RETNAMES['dasec'][1]
 del CSPYCE_DEFINITIONS['dasec']['done']
 
+################################################################################
+# These functions need to be passed a fixed-size array that's based on the value of
+# other arguments.  It is simpler to create those arrays in Python and pass them.
+################################################################################
+
+def dafgda(handle, begin, end):
+    size = abs(end - begin) + 1
```

Comment:
Why the `abs`? The function documentation says:

```
   1)  If `baddr' is zero or negative, the error SPICE(DAFNEGADDR)
       is signaled by a routine in the call tree of this routine.
   2)  If baddr > eaddr, the error SPICE(DAFBEGGTEND) is signaled by a
       routine in the call tree of this routine.
```

Wouldn't it be better just to use `size = 0 if begin <= 0 or begin > end else end - begin + 1` ?

Reply:
I copied code from spiceypy.  I don't know why they did that.  For begin <= 0, I'm happy to just let spice deal with that error.